user_interface/EbikeUIController/can.cpp

```diff
@@ -0,0 +1,121 @@
+#include "can.h"
+
+// CAN message flags 
+bool CAN_MB0_BikeStatus_Flag = false;
+bool CAN_MB1_MotorVoltages_Flag = false;
+bool CAN_MB2_RealCurrents_Flag = false;
+bool CAN_MB3_CalcValues_Flag = false;
+bool CAN_MB4_BmsSoc_Flag = false;
+
+// Function to initialize the CAN communication
+void canInit(FlexCAN_T4<CAN0, RX_SIZE_256, TX_SIZE_16>& can) {
+    Serial.begin(115200); // Set baud rate to 115200
+    can.begin(); // CAN initialization
+    can.setBaudRate(500000); // Set CAN baud rate to 500kbit
+    can.setMBFilter(REJECT_ALL);
+    can.setMBFilter(MB0, CAN_ID_BIKE_STATUS); // Set mailbox CAN ID filters
+    can.setMBFilter(MB1, CAN_ID_MOTOR_VOLTAGES);
+    can.setMBFilter(MB2, CAN_ID_MOTOR_REAL_CURRENTS);
+    can.setMBFilter(MB3, CAN_ID_MOTOR_CALC_VALUES);
+    can.setMBFilter(MB4, CAN_ID_BMS_SOC);
+    can.enableMBInterrupt(); // Enable all mailboxes to interrupts
+    can.onReceive(MB0, updateCAN_BikeStatus_Struct); // Set mailbox 0 to receive system parameters
+    can.onReceive(MB1, updateCAN_MotorVoltages_Struct); // Set mailbox 1 to receive system parameters
+    can.onReceive(MB2, updateCAN_RealCurrents_Struct); // Set mailbox 2 to receive system parameters
+    can.onReceive(MB3, updateCAN_CalcValues_Struct); // Set mailbox 3 to receive system parameters
+    can.onReceive(MB4, updateCAN_BmsSoc_Struct); // Set mailbox 4 to receive system parameters
+    can.setMB(MB5, TX); // Set mailbox 5 as transmit
+}
+
+//function used as callback when bike status message enters mailbox
+void updateCAN_BikeStatus_Struct(const CAN_message_t &msg) {
```

Comment:
For function naming, lets put the library/module first and the function name second.
something like canUpdateBikeStatus(...) would be better, up to you whether you want to make the library tag can or CAN or whatever else makes sense